apps/prairielearn/src/pages/publicAssessmentQuestionsPreview/publicAssessmentQuestionsPreview.ts

```diff
@@ -0,0 +1,70 @@
+import * as express from 'express';
+import asyncHandler from 'express-async-handler';
+import { z } from 'zod';
+
+import * as error from '@prairielearn/error';
+import { queryRow, loadSqlEquiv } from '@prairielearn/postgres';
+
+import { type Assessment, AssessmentSchema } from '../../lib/db-types.js';
+import { selectCourseById, selectCourseIdByInstanceId } from '../../models/course.js';
+import { selectAssessmentQuestions } from '../../models/questions.js';
+
+import { InstructorAssessmentQuestions } from './publicAssessmentQuestionsPreview.html.js';
+
+async function selectAssessmentById(assessment_id: string): Promise<Assessment> {
+  return await queryRow(
+    sql.select_assessment_by_id,
+    {
+      assessment_id,
+    },
+    AssessmentSchema,
+  );
+}
+
+const BooleanSchema = z.boolean();
+
+async function checkAssessmentPublic(assessment_id: string): Promise<boolean> {
+  const isPublic = await queryRow(sql.check_assessment_is_public, { assessment_id }, BooleanSchema);
+  return isPublic;
+}
+
+const router = express.Router();
+const sql = loadSqlEquiv(import.meta.url);
+
+router.get(
+  '/',
```

Comment:
I believe the public question page will 404 if a sharing name isn't set. Maybe we should do the same here? There's some weird behavior if one isn't set: the QIDs here show up as `@null/...`, and clicking on a question 404s.

Reply:
Throw 404 error if the course sharing name isn't set

~line 43:
```
if (course.sharing_name === null) {
    throw new error.HttpStatusError(404, 'Not Found');
}
```